src/UserContext.jsx

```diff
@@ -15,14 +21,20 @@ export const UserProvider = ({ children }) => {
 	const [refreshToken, setRefreshToken] = useState(null);
 	const [displayName, setDisplayName] = useState(null);
 	const [phoneNumber, setPhoneNumber] = useState(null);
+	const [userId, setUserId] = useState(null);
 	const [loggedIn, setLoggedIn] = useState(false);
 
+	useEffect(() => {
+		checkForDetails();
+	}, []); // check for details when component mounts ie app starts
+
 	const setUserDetails = useCallback(
 		(userDetails) => {
 			const base64Payload = userDetails.idToken.split(".")[1];
 			const decodedIdTokenPayload = JSON.parse(atob(base64Payload));
-			setDisplayName(decodedIdTokenPayload["custom:display_name"]);
+			setDisplayName(decodedIdTokenPayload["preferred_username"]);
```

Comment:
might not have to decode the id token for this now

Reply:
How come?

Reply:
ah wait, nevermind, I confused myself, we do still need to